<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>980</width>
    <height>874</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3DViewer</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_9" rowstretch="0,1" columnstretch="1,0">
    <item row="1" column="0">
     <widget class="GLWidget" name="GLwidget" native="true"/>
    </item>
    <item row="1" column="1">
     <layout class="QGridLayout" name="gridLayout_8">
      <item row="9" column="0">
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="2" column="1">
         <widget class="QPushButton" name="VertexColor">
          <property name="text">
           <string>Choose</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_22">
          <property name="text">
           <string>Color:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_21">
          <property name="text">
           <string>Type:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="VertexType">
          <item>
           <property name="text">
            <string>Circle</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Square</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QSlider" name="SizeVertexSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>20</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="SizeVertexSpinBox">
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>20.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.500000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_16">
          <property name="text">
           <string> Size:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Rotation</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="1">
         <widget class="QSlider" name="XRotSlider">
          <property name="minimum">
           <number>-360</number>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>X:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Y:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Z:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QDoubleSpinBox" name="XRotSpinBox">
          <property name="minimum">
           <double>-360.000000000000000</double>
          </property>
          <property name="maximum">
           <double>360.000000000000000</double>
          </property>
          <property name="value">
           <double>0.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSlider" name="YRotSlider">
          <property name="minimum">
           <number>-360</number>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QDoubleSpinBox" name="YRotSpinBox">
          <property name="minimum">
           <double>-360.000000000000000</double>
          </property>
          <property name="maximum">
           <double>360.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSlider" name="ZRotSlider">
          <property name="minimum">
           <number>-360</number>
          </property>
          <property name="maximum">
           <number>360</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QDoubleSpinBox" name="ZRotSpinBox">
          <property name="minimum">
           <double>-360.000000000000000</double>
          </property>
          <property name="maximum">
           <double>360.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="7" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_18" stretch="0,1">
        <item>
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>Scale:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="ScaleSpinBox">
          <property name="minimum">
           <double>0.010000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string>Edges</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="2">
         <widget class="QSlider" name="XMovSlider">
          <property name="minimum">
           <number>-100</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QSlider" name="YMovSlider">
          <property name="minimum">
           <number>-100</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Y:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Z:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>X:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QDoubleSpinBox" name="XMovSpinBox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::WheelFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::DefaultContextMenu</enum>
          </property>
          <property name="minimum">
           <double>-100.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="4">
         <widget class="QDoubleSpinBox" name="YMovSpinBox">
          <property name="minimum">
           <double>-100.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100.989999999999995</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QSlider" name="ZMovSlider">
          <property name="minimum">
           <number>-100</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="3" column="4">
         <widget class="QDoubleSpinBox" name="ZMovSpinBox">
          <property name="minimum">
           <double>-100.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Vertexes</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="14" column="0">
       <layout class="QGridLayout" name="gridLayout_7" columnstretch="0,1">
        <item row="0" column="0">
         <widget class="QLabel" name="label_24">
          <property name="text">
           <string>Color:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="BackgoundColor">
          <property name="text">
           <string>Choose</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Movement</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="13" column="0">
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>Background</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="1" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_21" stretch="0,0">
          <item>
           <widget class="QSlider" name="SizeEdgeSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>20</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="SizeEdgeSpinBox">
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>20.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.500000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="EdgeType">
          <item>
           <property name="text">
            <string>Solid</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Dashed</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QPushButton" name="EdgeColor">
          <property name="text">
           <string>Choose</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_19">
          <property name="text">
           <string>Color:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_23">
          <property name="text">
           <string>Type:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_20">
          <property name="text">
           <string>Size:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_15" stretch="0,1">
        <property name="spacing">
         <number>-1</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Projection:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="ProjectionType">
          <item>
           <property name="text">
            <string>Parallel</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Central</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="Reset">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0" colspan="2">
     <layout class="QFormLayout" name="formLayout_5">
      <property name="labelAlignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
      <property name="formAlignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>File:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="Filename">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Number of vertexes:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Number of facets:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="FacetsAmount">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>Number of edges:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="EdgesAmount">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="VertexesAmount">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>980</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave_As"/>
    <addaction name="actionSave_Gif"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open file</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As ...</string>
   </property>
  </action>
  <action name="actionSave_Gif">
   <property name="text">
    <string>Save Gif</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header location="global">view/glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
